src/componentWithName.js

```diff
@@ -16,7 +16,7 @@ function hasName(name, prop, propName, componentName) {
   }
 
   const { type } = prop;
-  if (getComponentName(type) !== name) {
+  if (!new RegExp(name).test(getComponentName(type))) {
```

Comment:
Yes, definitely. `new RegExp` is problematic, and it's not safe to pass unsanitized strings (that might have `[` and `]` in them, for example) to `new RegExp`.

Passing the responsibility on to the user to create the regex neatly sidesteps these issues.